yt/frontends/ramses/data_structures.py

```diff
@@ -109,37 +105,74 @@ def check_standard_files(folder, iout):
         ok &= (folder / f"info_{iout}.txt").is_file()
         return ok
 
-    @classmethod
-    def test_with_folder_name(cls, output_dir):
+    @staticmethod
+    def _match_output_and_group(
+        path: Path,
+    ) -> Tuple[Path, Optional[Path], Optional[str]]:
+        iout_match = OUTPUT_DIR_RE.match(path.name)
+
+        if not iout_match:
+            return path, None, None
+
+        # Two cases:
+        # 1. regular structure (all files in the output_XXXXX folder)
+        # 2. group structure (files spread in group_YYYYY folders)
+        group_dir: Optional[Path]
+
+        if iout_match.group(1) == "output":
+            # Case 1, unless the current folder contains a group folder itself
+            if (path / "group_00001").is_dir():
+                group_dir = path / "group_00001"
+            else:
+                group_dir = None
+            output_dir = path
+        else:
+            # Case 2, we need to extract output number from parent output folder
+            group_dir = path
+            output_dir = path.parent
+
+        # Match again to find output number
```

Comment:
It seems to me that we're pilling more and more unnecessary complexity here because `OUTPUT_DIR_RE` is _too_ flexible. Wouldn't it be a little simpler to have two separate expressions `OUTPUT_DIR_RE` and `GROUP_DIR_RE` ?